### PR TITLE
Cross-compile for Scala 2.12

### PR DESCRIPTION
**Pull request by rtyley**
We use this library in Ophan, and it would be nice to be able to upgrade to Scala 2.12!

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -15,12 +17,11 @@ lazy val core = project.settings(
   name := "marley",
   libraryDependencies ++= Seq(
     "org.apache.avro" % "avro" % "1.7.7",
-    "org.parboiled" %% "parboiled" % "2.1.0",
-    "org.typelevel" %% "macro-compat" % "1.1.1",
+    "org.parboiled" %% "parboiled" % "2.1.4",
     "org.scala-lang" % "scala-compiler" % scalaVersion.value % "provided",
-    compilerPlugin("org.scalamacros" % "paradise" % "2.1.0" cross CrossVersion.full),
-    "org.scalatest" %% "scalatest" % "2.2.5" % Test,
-    "org.scalacheck" %% "scalacheck" % "1.12.4" % Test
```
Comment (rtyley) [inline build.sbt]: These libraries (

File: build.sbt [hunk 2]
```diff
@@ -15,12 +17,11 @@ lazy val core = project.settings(
   name := "marley",
   libraryDependencies ++= Seq(
     "org.apache.avro" % "avro" % "1.7.7",
-    "org.parboiled" %% "parboiled" % "2.1.0",
-    "org.typelevel" %% "macro-compat" % "1.1.1",
```
Comment (rtyley) [inline build.sbt]: [`macro-compat`](https://github.com/milessabin/macro-compat) apparently is only useful for Scala 2.10 support, which I removed in e92f2cd6fa

File: build.sbt (hunk 1 above)
Comment (amyhughes) [reply build.sbt]: Nice, thanks for the link

**Comment by rtyley** [comment]
> Out of interest, is there a quick way of evaluating which dependencies in your project need updating for a particular Scala version?

I wasn't very sophisticated about it - I just updated `scalaVersion` to 2.12 and then addressed all of the fatal dependency errors sbt gave me, looking up the projects and finding out what the available artifact versions were in http://search.maven.org/ - perhaps the [Scaladex](https://index.scala-lang.org/search?scalaVersions=2.11&scalaVersions=2.12&q=parboiled) might have been a good complement to that.

But your question got me thinking, and I remembered that the best way to do this is probably to use the [`sbt-updates`](https://github.com/rtimush/sbt-updates) sbt plugin - once you install it, you can run a `dependencyUpdates` task, which gives output like this:

```
$ sbt dependencyUpdates
[info] Set current project to root (in build file:/home/roberto/guardian/marley/)
[info] No dependency updates found for root
[info] Found 3 dependency updates for marley
[info]   com.twitter:scrooge-core    : 4.12.0          -> 4.20.0 -> 18.2.0
[info]   org.apache.avro:avro        : 1.7.7           -> 1.8.2           
[info]   org.apache.thrift:libthrift : 0.9.2  -> 0.9.3 -> 0.11.0  
```

**Pull request merged**